embassy-sync/src/watch.rs

```diff
@@ -310,6 +310,18 @@ impl<M: RawMutex, T: Clone, const N: usize> Watch<M, T, N> {
         }
     }
 
+    /// Create a new `Watch` channel.
+    pub const fn new_with(data: Option<T>) -> Self {
```

Comment:
Rather than taking an option here, it should  pass the value T, since passing None it should use the regular new() instead.

Reply:
Makes sense to me! I've updated that 👍